### PR TITLE
feat(auth): add get/clear deviceMetadata mechanism

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

This PR adds a mechanism to get/clear `deviceMetadata` which is needed to start the device management flow when `Device tracking` is enabled in the `InitiateAuth` and `RespondToAuthChallenge` API calls. 

**Other changes:**
- Cleans up the sign-in tests. Previous sign-in tests were testing the same error cases. This change moves all sign-in error cases to a single file.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ashwinkumar6 — APPROVED** [review]

File: packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts [hunk 1]
```diff
@@ -38,115 +39,114 @@ export class DefaultTokenStore implements AuthTokenStore {
...
+			this.getAuthKeys().deviceMetadata
+		);
+	}
+
+	private getAuthKeys(): AuthKeys<keyof typeof AuthTokenStorageKeys> {
 		assertTokenProviderConfig(this.authConfig?.Cognito);
-		const name = 'Cognito'; // TODO(v6): update after API review for Amplify.configure
 		const authKeys = createKeysForAuthStorage(
-			name,
+			this.name,
```
Comment (HuiSF) [inline packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:146]: nit: can we memo createKeysForAuthStorage result with `userPoolClientId`. The `authKeys` is used a lot, we probably can avoid repetitive computing to save some CPU cycles.

File: packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts [hunk 2]
```diff
@@ -38,115 +39,114 @@ export class DefaultTokenStore implements AuthTokenStore {
...
-				(await this.getKeyValueStorage().getItem(authKeys.deviceMetadata)) ||
-					'{}'
-			);
-			const deviceMetadata =
-				Object.keys(newDeviceMetadata).length > 0
-					? newDeviceMetadata
-					: undefined;
+				(await this.getKeyValueStorage().getItem(
+					this.getAuthKeys().refreshToken
+				)) || undefined;
```
Comment (HuiSF) [inline packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:63]: nit: throughout

```suggestion
				(await this.getKeyValueStorage().getItem(
					this.getAuthKeys().refreshToken
				)) ?? undefined;
```

**Review by HuiSF — COMMENTED** [review]
Thanks for making the tokenOrchestrator injectable 👍

File: packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts (hunk 1 above)
Comment (israx) [reply packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:146]: good catch

**Review by ashwinkumar6 — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**